GCTC-NTGC/gc-digital-talent
# Feature/1464 add jest support

**Purpose**

To install/add jest and its related configuration to support unit testing typescript components


**How To Test?**

1.  check out the branch 
2.  run npm i
3. compile your code once ( npm run dev) 
4. npm run test  - should allow you to run the sample test for homepage
5. npm run test:watch - should allow you  run the tests while you are adding/fixing them 

**How To View Coverage?**
Coverage runs every `npm run test` execution.  Coverage report will be stored in below folder in html format.  Open it with browser to view the report. 

 ![Screen Shot 2021-12-30 at 3 08 47 PM](https://user-images.githubusercontent.com/89482910/147785361-b9439b50-a5cb-40d6-970d-8911db0ad690.png)

![Screen Shot 2021-12-30 at 2 36 05 PM](https://user-images.githubusercontent.com/89482910/147785370-def42a64-9dc0-4522-9b8d-c927a81d864c.png)
